src/redux/books/bookSlice.js

```diff
@@ -1,23 +1,107 @@
-import { createSlice } from '@reduxjs/toolkit';
-import BookItems from './BookItems';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+// import BookItems from './BookItems';
+import axios from 'axios';
+
+const apiUrl =
+  'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/WluyIk7BUwsnpiFriH9k/books/';
+
 const initialState = {
-  bookItems: BookItems,
+  bookItems: [],
+  isLoading: false,
 };
 
+export const getBooks = createAsyncThunk('books/getBooks', async (thunkAPI) => {
+  try {
+    // console.log(thunkAPI);
+    const resp = await axios.get(apiUrl);
+    // to fetch data from API with the desired format
+    const transformedBooks = Object.keys(resp.data).map((key) => {
+      const bookData = resp.data[key][0];
+      return {
+        item_id: key,
+        title: bookData.title,
+        author: bookData.author,
+        category: bookData.category,
+      };
+    });
+
+    console.log(transformedBooks);
+    return transformedBooks;
+  } catch (error) {
+    return thunkAPI.rejectWithValue(error);
+  }
+});
+
+export const postBooks = createAsyncThunk('books/postBooks', async (book, thunkAPI) => {
+  try {
+    const resp = await axios.post(apiUrl, book);
+
+    console.log(resp);
+    return resp.data;
+  } catch (error) {
+    return thunkAPI.rejectWithValue(error);
+  }
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async (id, thunkAPI) => {
+  try {
+    const resp = await axios.delete(`${apiUrl}${id}`);
+
+    console.log(resp);
+    return resp.data;
+  } catch (error) {
+    return thunkAPI.rejectWithValue(error);
+  }
+});
+
 const bookSlice = createSlice({
   name: 'books',
   initialState,
-  reducers: {
-    addBook: (state, action) => {
-      state.bookItems.push(action.payload);
-    },
-    removeBook: (state, action) => {
-      state.bookItems = state.bookItems.filter((book) => book.item_id !== action.payload);
-    },
+  reducers: {},
+  extraReducers(builder) {
+    builder
+      .addCase(getBooks.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(getBooks.fulfilled, (state, action) => {
+        console.log(action);
+        state.isLoading = false;
+        state.bookItems = action.payload;
+        console.log(action.payload);
+      })
+      .addCase(getBooks.rejected, (state, action) => {
+        console.log(action);
```

Comment:
- [ ] please remove all the console.logs they make your code look unprofessional and they slow down performance.